c/src/memory.rs

```diff
@@ -47,21 +47,21 @@ pub(super) fn release_string(str: String) -> *mut c_char {
     raw
 }
 
-pub(super) fn arc_into_raw<T>(t: Arc<T>) -> *const T {
+pub(super) fn unwrap_arc<T>(t: Arc<T>) -> *const T {
     let raw = Arc::into_raw(t);
-    trace!("Arc into raw of <{}> @ {:?}", std::any::type_name::<T>(), raw);
+    trace!("Unwrap arc to raw <{}> @ {:?}", std::any::type_name::<T>(), raw);
     raw
 }
 
-pub(super) fn arc_from_raw<T>(raw: *const T) -> Arc<T> {
-    trace!("Arc from raw of <{}> @ {:?}", std::any::type_name::<T>(), raw);
+pub(super) fn wrap_into_arc<T>(raw: *const T) -> Arc<T> {
+    trace!("Wrap raw into arc <{}> @ {:?}", std::any::type_name::<T>(), raw);
     unsafe { Arc::from_raw(raw) }
 }
 
 pub(super) fn release_arc<T>(raw: *const T) {
     trace!("Releasing ownership of arced <{}> @ {:?}", std::any::type_name::<T>(), raw);
     assert!(!raw.is_null());
-    drop(arc_from_raw(raw))
+    drop(wrap_into_arc(raw))
 }
```

Comment:
Naming here is messed up. We use `release_X` to mean "release ownership of `X` to SWIG"

It should be:
* `release_arc(Arc<T>) -> *const T` for giving up ownership (without decrementing),
* `take_arc(*const T) -> Arc<T>` to take ownership of the pointer back,
* and `decrement_arc(*const T) /* -> () */` (rather than `free_arc()` since it only decrements the counters and may not free) which is basically `drop(take_arc(t))`

Reply:
You are totally right. If I remember correctly, I didn't want to use `release` for `Arc`, but for some reason used it to free the memory. Thanks!